---
float64[] init_pos  
bool success